src/pilot/assign_role.py

```diff
@@ -1366,13 +1366,21 @@ def change_physical_disk_state_wait(
     change_state_result = drac_client.change_physical_disk_state(
         mode, controllers_to_physical_disk_ids)
 
+    realtime_controllers = []
     job_ids = []
     if change_state_result['commit_required_ids']:
         for controller_id in change_state_result['commit_required_ids']:
+            realtime = drac_client.is_realtime_supported(controller_id)
```

Comment:
Calling is_realtime_supported in this loop will beat up the iDRAC a little because it will result in len(change_state_result['commit_required_ids']) list_controller calls to the iDRAC.  I think it would be better to just call list_controllers once just before this loop and build a dict that maps controller_id to a boolean that indicates if the controller supports realtime mode.  This would turn the above line into a lookup into the dict by controller_id.  This would make it so we make fewer calls to the iDRAC, which will speed things up, and not put unnecessary stress on the iDRAC.

Reply:
Updated in new PR[#254](https://github.com/dsp-jetpack/JetPack/pull/254)